### PR TITLE
Update to Visual Studio 2019 and C# 8

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: NetSpell.SpellChecker/Dictionary/WordDictionary.cs [hunk 1]
```diff
@@ -14,8 +14,8 @@
 
 namespace NetSpell.SpellChecker.Dictionary
 {
-    using AffixRuleCollection = Dictionary<string, AffixRule>;
-    using PhoneticRuleCollection = List<PhoneticRule>;
+    using AffixRuleCollection = System.Collections.Generic.Dictionary<string, NetSpell.SpellChecker.Dictionary.Affix.AffixRule>;
+    using PhoneticRuleCollection = System.Collections.Generic.List<NetSpell.SpellChecker.Dictionary.Phonetic.PhoneticRule>;
```
Comment (RussKie) [inline NetSpell.SpellChecker/Dictionary/WordDictionary.cs]: Why have these changed?

File: Build/stylecop.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
         "namingRules": {
             "allowedHungarianPrefixes": [
                 "ui", "x", "y", "my", "ls", "am", "id", "ie", "vs", "up"
-            ]
+            ],
+            "tupleElementNameCasing": "camelCase"
```
Comment (RussKie) [inline Build/stylecop.json]: I am not overly fond of these change, callsites look worse.
Besides it seems like a noisy change

Comment (sharwell) [reply Build/stylecop.json]: Both styles were in use before, so this was just a consolidation.

File: NetSpell.SpellChecker/Dictionary/WordDictionary.cs (hunk 1 above)
Comment (sharwell) [reply NetSpell.SpellChecker/Dictionary/WordDictionary.cs]: A probable bug in the analyzer, but seemed to have low impact overall.

Comment (RussKie) [reply NetSpell.SpellChecker/Dictionary/WordDictionary.cs]: Please revert. I can live with the rest.

File: Build/stylecop.json (hunk 2 above)
Comment (RussKie) [reply Build/stylecop.json]: Pascal case at callsites feels more C#

File: NetSpell.SpellChecker/Dictionary/WordDictionary.cs (hunk 1 above)
Comment (sharwell) [reply NetSpell.SpellChecker/Dictionary/WordDictionary.cs]: Done

File: Build/stylecop.json (hunk 2 above)
Comment (sharwell) [reply Build/stylecop.json]: I went through again and confirmed that camelCase is the dominant style in the code base (by far). It follows the view that tuples are more like a bag of parameters than a type with named fields.

**Pull request merged**